server/src/main/scala/services/arrivals/ArrivalDataSanitiser.scala

```diff
@@ -0,0 +1,38 @@
+package services.arrivals
+
+import drt.shared.Arrival
+
+case class ArrivalDataSanitiser(
+                                 maybeEstimatedThresholdHours: Option[Int],
+                                 maybeTaxiThresholdMinutes: Option[Int]) {
+
+  def maybeEstThresholdMillis: Option[Long] = maybeEstimatedThresholdHours.map(h => h * 60 * 60 * 1000L)
+
+  def maybeTaxiThresholdMillis: Option[Long] = maybeTaxiThresholdMinutes.map(m => m * 60 * 1000L)
+
+  def withSaneEstimates(arrival: Arrival): Arrival = withSaneEstimatedChox(withSaneEstimatedTouchDown(arrival))
+
+  def withSaneEstimatedTouchDown(arrival: Arrival): Arrival =
+    (maybeEstThresholdMillis, arrival.Estimated, arrival.Actual) match {
+      case (Some(threshold), Some(est), _) if (Math.abs(est - arrival.Scheduled) > threshold) =>
+        arrival.copy(Estimated = None)
+      case (_, Some(est), Some(act)) if est != act =>
+        arrival.copy(Estimated = arrival.Actual)
```

Comment:
Is this solving an actual issue? I see that some flights have an actual touchdown up to a few minutes out from the estimated time, but that's presumably fine as it just reflects that the last estimate wasn't 100% spot on. Are there cases where using the actual as the estimate is really desirable?

Reply:
happy to take it out! just seems to me like the estimate should be the same as the actual once the plane has landed - you are right though, there's no issue here as the estimated time won't be used for anything once the actual arrival has been logged.

Reply:
cool. i think i'd be inclined to leave the estimated as is if it's not causing an issue